src/Service/Linux/OrbitService.cpp

```diff
@@ -129,19 +132,55 @@ static std::unique_ptr<ProducerSideServer> BuildAndStartProducerSideServer() {
     return nullptr;
   }
 
+  std::string unix_socket_path(orbit_producer_side_channel::kProducerSideUnixDomainSocketPath);
   auto producer_side_server = std::make_unique<ProducerSideServer>();
-  LOG("Starting producer-side server at %s",
-      orbit_producer_side_channel::kProducerSideUnixDomainSocketPath);
-  if (!producer_side_server->BuildAndStart(
-          orbit_producer_side_channel::kProducerSideUnixDomainSocketPath)) {
+  LOG("Starting producer-side server at %s", unix_socket_path);
+  std::string uri = absl::StrFormat("unix:%s", unix_socket_path);
+  if (!producer_side_server->BuildAndStart(uri)) {
     ERROR("Unable to start producer-side server");
     return nullptr;
   }
+
+  // When OrbitService runs as root, also allow non-root producers
+  // (e.g., the game) to communicate over the Unix domain socket.
+  if (chmod(unix_socket_path.c_str(), 0777) != 0) {
+    ERROR("Changing mode bits to 777 of \"%s\": %s", unix_socket_path, SafeStrerror(errno));
+    producer_side_server->ShutdownAndWait();
+    return nullptr;
+  }
+
   LOG("Producer-side server is running");
   return producer_side_server;
 }
 
-int OrbitService::Run(std::atomic<bool>* exit_requested) {
+std::atomic<bool> exit_requested = false;
+
+void SigintHandler(int signum) {
+  if (signum == SIGINT) {
+    exit_requested = true;
+  }
+}
+
+void InstallSigintHandler() {
+  struct sigaction act {};
+  act.sa_handler = SigintHandler;
+  sigemptyset(&act.sa_mask);
+  act.sa_flags = 0;
+  act.sa_restorer = nullptr;
+  sigaction(SIGINT, &act, nullptr);
+}
+
+}  // namespace
+
+std::string OrbitService::GetLogFilePath() {
```

Comment:
In my opinion this was also clearer in main as it doesn't really have any association with the class. Sure, you avoid an `#ifdef` in `main.cpp`, but `if(WIN32)` is just an `#ifdef` in a different form.